scrapper.py

```diff
@@ -17,47 +34,152 @@ class NumberOfArticlesOutOfRangeError(Exception):
     """
-    Total number of articles to parse in not integer
+    Total number of articles to parse is not integer  # in|is !typo
+    """
+
+
+class BadStatusCode(Exception):
+    """
+    Request did not give valid response
     """
 
 
 class Crawler:
     """
     Crawler implementation
     """
-    def __init__(self, seed_urls, max_articles: int):
-        pass
+
+    def __init__(self, seed_urls: List[str], max_articles: int):
+        self.seed_urls = seed_urls
+        self.total_max_articles = max_articles
+        self.urls = []
 
     def _extract_url(self, article_bs):
-        pass
+        contents_bs = article_bs.select('div.content-cont-col-text a')
+        links = ['https://alp.iling.spb.ru/' + content['href'][2:] for content in contents_bs]
+        self.urls.extend(links)
 
     def find_articles(self):
         """
         Finds articles
         """
-        pass
+        for url in self.seed_urls:
+            page = obtain_page(url)
+            soup = BeautifulSoup(page, 'lxml')
+            self._extract_url(soup)
+            sleep(random.uniform(2, 4))
+            if len(self.urls) < self.total_max_articles:
+                continue
+            self.urls = self.urls[:self.total_max_articles]
 
     def get_search_urls(self):
         """
         Returns seed_urls param
         """
-        pass
+        return self.seed_urls
+
+
+class HTMLParser:
+    """
+    Parser implementation
+    """
+
+    def __init__(self, article_url, article_id):
+        self.article_url = article_url
+        self.article_id = article_id
+        self.article = Article(article_url, article_id)
+
+    def _fill_article_with_text(self, article_bs):
+        pdf_url = 'https://alp.iling.spb.ru/' + article_bs.select_one('div.article-download-r a')['href'][5:]
+        pdf_file = PDFRawFile(pdf_url, self.article_id)
+        pdf_file.download()
+        self.article.text = pdf_file.get_text()
+
+    def _fill_article_with_meta_information(self, article_bs):
+        self.article.title = article_bs.select_one('h1.article-h1').text
+
+        author = article_bs.select_one('span.author-name').text.replace('\n', ' ')
+        if not author:
+            self.article.author = 'NOT FOUND'
+        self.article.author = author
+
+        issue = re.search(r'(?P<volume>x.+)(?P<part>\d)(?=/)', self.article_url)
+        date_raw = f"01.{ISSUE_MONTHS.get(issue.group('part'))}.{ISSUE_YEARS.get(issue.group('volume'))}"
+        article_date = datetime.datetime.strptime(date_raw, '%d.%m.%Y')
+        self.article.date = article_date
+
+        self.article.topics = article_bs.select_one('div.kw-tags').text.split(', ')
+
+    def parse(self):
+        page = obtain_page(self.article_url)
+        article_bs = BeautifulSoup(page, 'lxml')
+        self._fill_article_with_text(article_bs)
+        self._fill_article_with_meta_information(article_bs)
+        return self.article
+
+
+def obtain_page(url):
```

Comment:
why do you add such a method? why it is not included into class?

Reply:
To avoid dupplicating code patterns. It is not a class method because it should be within the reach of both Crawler and HTML parser.